mmv1/third_party/terraform/services/compute/data_source_google_compute_forwarding_rules.go

```diff
@@ -0,0 +1,101 @@
+package compute
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceGoogleComputeForwardingRules() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleComputeForwardingRulesRead,
+
+		Schema: map[string]*schema.Schema{
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"rules": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeMap},
```

Comment:
why not reuse the resource schema of the existing forwarding rule schema? similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/sql/data_source_sql_databases.go#L31

I'm not sure a string map will be the best experience for users.

Reply:
I have updated the return type to be the Forwarding Rules schema now.